libtenzir/builtins/contexts/lookup_table.cpp

```diff
@@ -502,19 +510,87 @@ class ctx final : public virtual context {
       }
       return result;
     };
-    return update_result{
+    return context_update_result{
       .update_info = show(),
       .make_query = std::move(query_f),
     };
   }
 
+  auto update2(const table_slice& events, const context_update_args& args,
+               session ctx) -> failure_or<context_update_result> override {
+    for (const auto& timeout :
+         {args.create_timeout, args.write_timeout, args.read_timeout}) {
+      if (timeout) {
+        diagnostic::warning("unsupported option for bloom-filter context")
+          .primary(*timeout)
+          .emit(ctx);
+      }
+    }
+    auto keys = eval(args.key, events, ctx);
+    auto key_values_list = list{};
+    const auto update_entry
+      = [&, now = time::clock::now()](bool created, value_data& entry,
+                                      record context) {
+          entry.raw_data = std::move(context);
+          if (created and args.create_timeout) {
+            entry.create_timeout = now + args.create_timeout->inner;
+          }
+          if (args.write_timeout) {
+            entry.write_timeout = now + args.write_timeout->inner;
```

Comment:
This seems to update unconditionally when the timeout is present. Maybe I don't have the full picture, but I thought the update should only takes place when we actually modify (= *write*) a value.

Reply:
This code is called as part of `context::update` for a lookup table. The variable `created` is set to true when the key did not previously exist, so that's when we update the create timeout. We always update the write and read timeout here as well. We additionally update the read timeout in `context::enrich` and `context::lookup` (that's the `apply` function below).